crates/python_extractor/src/lib.rs

```diff
@@ -44,20 +101,47 @@ pub async fn extract_python(
         let mut file = std::fs::File::open(&input_file).with_context(|| {
             format!(
                 "While attempting to open file: {:?
-    }",
+        }",
                 input_file
             )
         })?;
+
         let mut buffer = Vec::new();
         file.read_to_end(&mut buffer).with_context(|| {
             format!(
                 "While attempting to read file: {:?
-    }",
+        }",
+                input_file
+            )
+        })?;
+
+        let mut reader = std::io::BufReader::new(&buffer[..]);
+        let mut first_line = String::new();
+        reader.read_line(&mut first_line).with_context(|| {
+            format!(
+                "While attempting to read first line from file to check for encoding: {:?
+            }",
                 input_file
             )
         })?;
 
-        let input_str = String::from_utf8_lossy(&buffer).into_owned();
+        let encoding = if first_line.starts_with("# coding:") {
```

Comment:
seems like we went pretty far here, what about using the regex from the standard something like `.*#.*\bcoding[=:]\s*([-\w.]+)` using the regex crate?